core/src/main/java/com/graphhopper/util/GHUtility.java

```diff
@@ -723,6 +727,20 @@ public static long calcMillisWithTurnMillis(Weighting weighting, EdgeIteratorSta
         }
     }
 
+    public static void runConcurrently(List<Callable<String>> callables, int threads) {
+        ExecutorService executorService = Executors.newFixedThreadPool(threads);
+        ExecutorCompletionService<String> completionService = new ExecutorCompletionService<>(executorService);
+        callables.forEach(completionService::submit);
+        executorService.shutdown();
+        try {
+            for (int i = 0; i < callables.size(); i++)
+                completionService.take().get();
+        } catch (Exception e) {
+            executorService.shutdownNow();
+            throw new RuntimeException(e);
+        }
```

Comment:
```suggestion
        try {
            for (int i = 0; i < callables.size(); i++)
                completionService.take().get();
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            throw new RuntimeException(e);
        } catch (ExecutionException e) {
            throw new RuntimeException(e);
        } finally {
            executorService.shutdown();
        }
```

Reply:
I tried to simply keep it as it was before. There is a long, possibly related discussion about this stuff here: #980